nautobot/extras/forms.py

```diff
@@ -771,6 +782,53 @@ def requires_input(self):
         return bool(len(self.fields) > 1)
 
 
+class JobScheduleForm(BootstrapMixin, forms.Form):
+    """
+    This form is rendered alongside the JobForm but deals specifically with the fields needed to either
+    execute the job immediately, or schedule it for later. Each field name is prefixed with an underscore
+    because in the POST body, they share a namespace with the JobForm which includes fields defined by the
+    job author, so the underscore prefix helps to avoid name collisions.
+    """
+
+    _schedule_type = forms.ChoiceField(
+        choices=choices.JobExecutionType,
+        help_text="The job can either run immediately, once in the future, or on a recurring schedule.",
+        label="Type",
+    )
+    _schedule_name = forms.CharField(
+        required=False,
+        label="Schedule name",
+        help_text="Name for the job schedule.",
+    )
+    _schedule_start_time = forms.DateTimeField(
+        required=False,
+        label="Starting date and time",
+        widget=DateTimePicker(),
+    )
+
+    def clean(self):
+        """
+        Validate all required information is present if the job needs to be scheduled
+        """
+        cleaned_data = super().clean()
+
+        if cleaned_data["_schedule_type"] != choices.JobExecutionType.TYPE_IMMEDIATELY:
+            if not cleaned_data["_schedule_name"]:
+                raise ValidationError({"_schedule_name": "Please provide a name for the job schedule."})
+
+            if ScheduledJob.objects.filter(name=cleaned_data["_schedule_name"]).exists():
+                # django_celery_beat.models.PeriodicTask enforces unique values for the name field, so we need to check
+                # for existing instances with the same name.
+                raise ValidationError({"_schedule_name": "Scheduled job with this name already exists!"})
+
+            if not cleaned_data["_schedule_start_time"] or cleaned_data["_schedule_start_time"] < timezone.now():
```

Comment:
Same comment here about defining some minimal window around `timezone.now()` for scheduling.

Reply:
I defined `JobSchedule.earliest_possible_time()` as a function for checking this (it’s now+15s). I’m not 100% comfortable with the name, but I think it captures the  intent well enough.